decorators_library/decorators.py

```diff
@@ -1 +1,85 @@
 # implement your decorators here.
+import time
+# from . import TimeoutError
+class TimeoutError(Exception):
+    def __init__(self, value):
+        self.value = value
+    def __str__(self):
+        return repr(self.value)
+
+# def timeout(func, t):
+#     start_time = time.time()
+#     def new_func(t):
+#         if time.time() - start_time > t:
+#             res = func()
+#             raise TimeoutError("Function call timed out")
+#         return res
+#     return new_func
+
+class timeout(object):
+    def __init__(self, t):
+        self.t = t
+    
+    def __call__(self, func):
+        start_time = time.time()
+        func()
+        total_time = time.time() - start_time
```

Comment:
Looks good!

After some googling, we found [helpful code](https://docs.python.org/3/library/signal.html#example) in the `signal` module that allows you to interrupt execution after a time limit has been set.  This would make the decorator useful in the event that it decorates a function that might hang indefinitely or for a very long time.